### PR TITLE
po/fix fast img reload

### LINKED ISSUE #13893: Frequent crashes
**Describe the bug/issue**
Frequesnt crashes during editing. After restart, the same images don't cause a crash.
Backtrace is empty; I'm attaching the log captured from the console.

[dt.log](https://github.com/darktable-org/darktable/files/10948583/dt.log)
The referenced log file, /tmp/darktable_bt_5J9901.txt, is empty, save for:

> this is darktable 4.3.0+1130~gf50cf5ba68 reporting a segfault:

**To Reproduce**
Editing images, switching between images.

With the files in:
https://tech.kovacs-telekes.org/files/darktable-crash-13893/
Switching between the files with space, without editing, triggers the crash. Sometimes I can get through the whole album, sometimes it crashes on a random image. I try to switch between the images as quickly as possible, always going to the next.

I could try a bisect, or provide more info if the attached log does not provide sufficient info. **Bisect pending.**

Self-compiled, Ubuntu 22.10, 'Release' mode. Revision f50cf5ba68. OpenCL on/off does not matter.

### PR DESCRIPTION
**Pull request by TurboGit**
Attempt to fix #13893.

### CODE REVIEW

File: src/views/darkroom.c [hunk 1]
```diff
@@ -812,7 +812,11 @@ static void _fire_darkroom_image_loaded_event(const bool clean, const int32_t im
 static void _dev_change_image(dt_develop_t *dev, const int32_t imgid)
 {
   // stop crazy users from sleeping on key-repeat spacebar:
-  if(dev->image_loading) return;
+  dt_pthread_mutex_lock(&dev->pipe_mutex);
+  const gboolean is_loading = dev->image_loading;
```
Comment (ralfbrown) [inline src/views/darkroom.c:816]: Mutexes are the same as atomics - *every* access to a variable needs to be protected by the same lock or you can have data races.

If dt were written entirely in C++, we could just declare the variable atomic and have the compiler ensure that all accesses use atomic instructions, but for plain C the programmer has to explicitly change all of the accesses.

Comment (TurboGit) [reply src/views/darkroom.c:816]: > Mutexes are the same as atomics - _every_ access to a variable needs to be protected by the same lock or you can have data races.

Yes, and that's the case. During the run of the pipe in the proper section in `dt_dev_process_image_job()` the lock is taken. So this patch ensure that we do not try to reload an image until the previous one has been loaded.

Comment (TurboGit) [reply src/views/darkroom.c:816]: But this is maybe too conservative anyway and I'm currently testing without this part anyway.

**Comment by TurboGit** [comment]
My testing shows that I cannot crash after removing the darkroom code. So now the patch is simpler and still fixes the issue.

File: src/views/darkroom.c (hunk 1 above)
Comment (TurboGit) [reply src/views/darkroom.c]: New version without this pushed now. This part was indeed not needed after all.

Comment (TurboGit) [reply src/views/darkroom.c]: And thanks for the review!

**Pull request merged**